src/main/scala/com/socrata/regioncoder/RegionCoder.scala

```diff
@@ -41,11 +42,38 @@ trait RegionCoder {
     }
   }
 
+  protected def regionCodeByTransform(resourceName: String,
+                                      idFeatureColumn: String,
+                                      labelColumnToReturn: String,
+                                      points: Seq[Seq[Double]]): Future[Seq[Option[Any]]] = {
+    val geoPoints = points.map { case Seq(x, y) => builder.Point(x, y) }
+    val partitions = pointsToPartitions(geoPoints)
+    val indexStringMap = labelCache.constructHashMap(
+      sodaFountain,
+      resourceName,
+      labelColumnToReturn,
+      idFeatureColumn
+    )
+    // Map unique partitions to SpatialIndices, fetching them in parallel using Futures
+    // Now we have a Seq[Future[Envelope -> SpatialIndex]]
+    val indexFutures = partitions.toSet.map { partEnvelope: Envelope =>
+      spatialCache.getFromSoda(sodaFountain, resourceName, idFeatureColumn, Some(partEnvelope))
+        .map(partEnvelope -> _)
+    }
+    // Turn sequence of futures into one Future[Map[Envelope -> SpatialIndex]]
+    // which will be done when all the indices/partitions have been fetched
+    Future.sequence(indexFutures).map(_.toMap).map { envToIndex =>
```

Comment:
are we bounding this parallelism at all? if not, we should

Reply:
Do you have an example of somewhere else we do this, I could crib off?

Reply:
Assuming the PR that @rjmac merged in yesterday addressed this, I went ahead and made a similar change to LabelCache.